src/scheduler/work_bucket.rs

```diff
@@ -232,6 +232,8 @@ pub enum WorkBucketStage {
     /// Clear the VO bit metadata.  Mainly used by ImmixSpace.
     #[cfg(feature = "vo_bit")]
     ClearVOBits,
+    /// Compute the transtive closure following only strong references, making sure that objects do not move
+    ImmovableClosure,
```

Comment:
One `ImmovableClosure` may not be enough.  We need two.

```suggestion
    /// Compute the transtive closure starting from t-pin (transitively pinning) roots, following only strong references.  No objects in this transitive closure may move.
    TPinClosure,
    /// Trace nt-pin (non-transitively pinning) roots, but do not compute transitive closure.  Nt-pin roots may not move, but their children may move.
    NTPinTrace,
```

One reason is that black (nt-pin) roots must be traced before any other non-pinning edges.  Otherwise, the object may have been moved before we pin it.  If we put `ProcessRootNode` work packets for black roots into the `Closure` bucket, other non-pinning packets in the `Closure` buckets may be executed before this.  So `ProcessRootNode` work packets for black roots needs to be executed before the `Closure` bucket.

The other reason is that `ProcessRootNode` work packets for black roots must be executed after all objects in transitive closure of red (t-pin) roots are visited.

1.  An object may be pointed by both a black root and a red root.  Suppose we have two variables, one C local variable and a Julia local variable.  Then the C variable will be red and the Julia variable will be black. (Correct me if I am wrong.)  It is possible that both variables point to the same object. When we scan stacks, both variables will be visited, and they will be added to two `ProcessRootNode` work packets, one for red roots, and the other for black roots.
2.  If both work packets are in the same bucket, the black (nt-pin) packet may be executed first.  It marks the object.  Then the read packet is executed.  When visiting the same object, it finds the object is already marked, and will not enqueue its children into the red transitive closure.  That will cause the children of that object not to be pinned, even though it needs transitive pinning.

For those reasons, we need two buckets: `TPinClosure` and `NTPinTrace` before the `Closure` bucket.  The `NTPinTrace` bucket is not named `*****Closure` because the children of black (nt-pin) roots belong to the regular non-pinned transitive closure.